src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -2222,6 +2222,59 @@ namespace {
     EXPECT_EQ(0, countNonzero(activeColumns));
   }
 
+
+  TEST(SpatialPoolerTest, testConstructorInitParamsUnstable)
+  {
+  /** this test exposes bug where c++ SP (wrongly) produces
+  different output for the same input. 
+  XXX sensitive - marks (empirically!) discovered set of parameter 
+  that produce the err behavior; changing any of the XXX params 
+  may cause the SP to behave "normally"
+  */
+    SpatialPooler sp{std::vector<UInt>{10} /* input*/, std::vector<UInt>{2048}/* SP output cols XXX sensitive*/, 
+                        /*pot radius*/ 20, //each col sees
+                        /*pot pct*/ 0.5, //XXX sensitive
+                        /*global inhibition*/ false, //XXX sensitive
+                       /*Real localAreaDensity=*/0.02, //2% active cols
+                       /*UInt numActiveColumnsPerInhArea=*/0, //mutex with above ^^ //XXX sensitive
+};
+
+    vector<UInt> input = { 1, 1, 0, 0, 1, 1, 0, 0, 1, 1};
+    vector<UInt> out1(sp.getNumColumns(), 0); 
+    vector<UInt> out2(sp.getNumColumns(), 0); 
+    sp.compute(input.data(), true, out1.data());
+    sp.compute(input.data(), true, out2.data());
+    EXPECT_EQ(out1, out2); //not necessary with the check in SP initialize(), but keep here as example
+}
+
+
+  TEST(SpatialPoolerTest, testDifferentConstructorVsSetterBehavior)
+  {
+  /** this test exposes wrong behavior, where SP created via constructor
+  behaves differently to a SP via setters (setXXX()), both with the same
+  params. 
```

Comment:
Test case: different behavior when setters vs constructor is used for initialization